adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -251,6 +256,14 @@ public static AuthenticationResult processUIResponseParams(Map<String, String> r
 
         } else if (response.containsKey(AuthenticationConstants.OAuth2.CODE)) {
             result = new AuthenticationResult(response.get(AuthenticationConstants.OAuth2.CODE));
+            final String cloudInstanceName = response.get(AuthenticationConstants.OAuth2.CLOUD_INSTANCE_NAME);
+            if (!StringExtensions.isNullOrBlank(cloudInstanceName)) {
+                final StringBuilder cloudInstanceUrl = new StringBuilder("https://login.")
+                        .append(cloudInstanceName)
+                        .append("/common");
```

Comment:
you should not hard code common as a tenant name. This will result in a bug for scenarios where developer passes https://login.microsoftonline.com/mysovereigntenant.com to the API and guest user will not be able to sign in because your code will attempt to redeem auth code against /common instead of guest tenant.